modules/renter/negotiate.go

```diff
@@ -2,233 +2,328 @@ package renter
 
 import (
 	"errors"
-	"io"
+	//"io"
 	"net"
-	"os"
-	"sync/atomic"
+	"sync"
 	"time"
 
-	"github.com/NebulousLabs/Sia/build"
 	"github.com/NebulousLabs/Sia/crypto"
 	"github.com/NebulousLabs/Sia/encoding"
 	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/types"
 )
 
-const (
-	defaultWindowSize = 288 // 48 Hours
-)
+// A hostUploader uploads pieces to a host. It implements the uploader interface.
+type hostUploader struct {
+	// constants
+	settings         modules.HostSettings
+	masterKey        crypto.TwofishKey
+	unlockConditions types.UnlockConditions // renter needs to save this!
```

Comment:
is this right? If so, the UnlockConditions include both public keys, so that's convenient. At the moment, there's no support for revising an "old" contract (you only revise during upload; no resuming). Even though we don't need to save it yet, we may want to in order to avoid breaking compatibility later. (Although that would also mean saving all the fields necessary for resumable uploads/downloads, which may be a bit trickier...)